src/components/Text/Text.tsx

```diff
@@ -1,32 +1,69 @@
 import clsx from 'clsx';
-import React, { ReactNode } from 'react';
+import React, { forwardRef, ForwardedRef } from 'react';
 import styles from './Text.module.css';
 
-export interface Props {
-  /**
-   * The rendered tag name of the Heading. The tag name should always relate to the [document outline](http://html5doctor.com/outlines/) and a tag name should never be chosen for its default aesthetic qualities. If a specific style is desired, use the `size` prop to manipulate the Heading style.
-   */
-  as: 'p' | 'span';
-  /**
-   * The child node(s)
-   */
-  children?: ReactNode;
+export type Size = 'body' | 'sm' | 'md' | 'lg' | 'xs' | 'caption' | 'overline';
+
+export type Color =
+  | 'alert'
+  | 'base'
+  | 'brand'
+  | 'info'
+  | 'inherit'
+  | 'neutral'
+  | 'success'
+  | 'warning'
+  | 'white';
+
+export type Props = {
   /**
-   * CSS class names that can be appended to the component.
+   * Controls whether to render text inline (defaults to "p");
    */
+  as?: 'p' | 'span';
+  children: React.ReactNode;
   className?: string;
-}
+  color?: Color;
+  size?: Size;
+  tabIndex?: number;
+  weight?: 'bold' | 'normal' | null;
+} & React.HTMLAttributes<HTMLElement>;
 
 /**
- * Primary UI component for user interaction
+ * ```ts
+ * import {Text} from "@chanzuckerberg/eds";
+ * ```
  */
-export const Text = ({ as = 'p', className, children, ...other }: Props) => {
-  const componentClassName = clsx(styles['text'], className, {});
-  const TagName = as;
-
-  return (
-    <TagName className={componentClassName} {...other}>
-      {children}
-    </TagName>
-  );
-};
+export const Text = forwardRef(
+  (
+    {
+      as = 'p',
+      children,
+      className,
+      color,
+      size = 'body',
+      weight,
+      /**
+       * Components that wrap typography sometimes requires props such as event handlers
+       * to be passed down into the element. One example is the tooltip component.  It
+       * attaches a onHover and onFocus event to the element to determine when to
+       * trigger the overlay.
+       */ ...other
+    }: Props,
+    ref?: ForwardedRef<HTMLParagraphElement>, // Setting as HTMLParagraphElement to satisfy TS, but unit test covers both span and p cases for sanity
```

Comment:
Ref tested with both `span` and `p`

Reply:
nit: I don't think you have to type the `ref` as optional